warp10/src/main/java/io/warp10/script/ext/http/HTTP.java

```diff
@@ -0,0 +1,449 @@
+            throw new WarpScriptException(getName() + " expects a list with two items, username and password, in argument " + authInfo + ".");
+          }
+
+          if (!(authInfo.get(0) instanceof String)) {
+            throw new WarpScriptException(getName() + " expects a STRING username when using basic authentication.");
+          }
+
+          if (!(authInfo.get(1) instanceof String)) {
+            throw new WarpScriptException(getName() + " expects a STRING password when using basic authentication.");
+          }
+
+          String userInfo = authInfo.get(0) + ":" + authInfo.get(1);
+          String basicAuth = "Basic " + Base64.encodeBase64String(userInfo.getBytes(StandardCharsets.UTF_8));
+          additionalHeaders =  new HashMap<Object, Object>();
+          additionalHeaders.put("Authorization", basicAuth);
+        }
+
+        headers.putAll(additionalHeaders);
+      }
+
+      for (Map.Entry<Object, Object> prop: headers.entrySet()) {
+        conn.setRequestProperty(String.valueOf(prop.getKey()), String.valueOf(prop.getValue()));
+      }
+
+      conn.setDoInput(true);
+      conn.setRequestMethod(method.toUpperCase());
+
+      //
+      // Set body
+      //
+
+      if (body instanceof String) {
+        String bodyS = (String) body;
+        conn.setDoOutput(bodyS.length() > 0);
+        if (bodyS.length() > 0) {
+          try (OutputStream os = conn.getOutputStream()) {
+            os.write(bodyS.getBytes(StandardCharsets.UTF_8));
+          }
+        }
+
+      } else if (body instanceof byte[]) {
+        byte[] bodyB = (byte[]) body;
+        conn.setDoOutput(bodyB.length > 0);
+        if (bodyB.length > 0) {
+          try (OutputStream os = conn.getOutputStream()) {
+            os.write(bodyB);
+          }
+        }
+
+      } else if (null != body) {
+        throw new WarpScriptException(getName() + " expects the body of the request to be a STRING or BYTES object.");
+      }
+
+      //
+      // Form response
+      //
+
+      res.put(STATUS_CODE, conn.getResponseCode());
+      Map<String, List<String>> hdrs = conn.getHeaderFields();
+
+      if (hdrs.containsKey(null)) {
+        List<String> statusMsg = hdrs.get(null);
+        if (statusMsg.size() > 0) {
+          res.put(STATUS_MESSAGE, statusMsg.get(0));
+        } else {
+          res.put(STATUS_MESSAGE, "");
+        }
+      } else {
+        res.put(STATUS_MESSAGE, "");
+      }
+
+      //
+      // Make the headers map modifiable
+      //
+
+      hdrs = new LinkedHashMap<String, List<String>>(hdrs);
+      hdrs.remove(null);
+
+      res.put(RESPONSE_HEADERS, hdrs);
+
+      //
+      // Read response
+      //
+
+      InputStream in = null;
+
+      // When there is an error (response code is 404 for instance), body is in the error stream.
+      try {
+        in = conn.getInputStream();
+      } catch (IOException ioe) {
+        in = conn.getErrorStream();
+      }
+
+      if (chunkSize <= 0) {
```

Comment:
what semantics does a negatice chunkSize have?

Reply:
none other than not using chunk option